examples/demo.citty.ts

```diff
@@ -0,0 +1,132 @@
+import { defineCommand, createMain } from 'citty';
+import tab from '../src/citty';
+
+const main = defineCommand({
+  meta: {
+    name: 'vite',
+    version: '0.0.0',
+    description: 'Vite CLI',
+  },
+  args: {
+    config: {
+      type: 'string',
+      description: 'Use specified config file',
+      alias: 'c',
+    },
+    mode: {
+      type: 'string',
+      description: 'Set env mode',
+      alias: 'm',
+    },
+    logLevel: {
+      type: 'string',
+      description: 'info | warn | error | silent',
+      alias: 'l',
+    },
+  },
+  subCommands: {
+    dev: defineCommand({
+      meta: {
+        name: 'dev',
+        description: 'Start dev server',
+      },
+      args: {
+        host: {
+          type: 'string',
+          description: 'Specify hostname',
+          alias: 'H',
+        },
+        port: {
+          type: 'string',
+          description: 'Specify port',
+          alias: 'p',
+        },
+      },
+      run: () => {},
+    }),
+    build: defineCommand({
+      meta: {
+        name: 'build',
+        description: 'Build project',
+      },
+      run: () => {},
+    }),
+    lint: defineCommand({
+      meta: {
+        name: 'lint',
+        description: 'Lint project',
+      },
+      args: {
+        files: {
+          type: 'positional',
+          description: 'Files to lint',
+          required: false,
+        },
+      },
+      run: () => {},
+    }),
+  },
+  run: () => {},
+});
+
+const completion = await tab(main);
```

Comment:
The code in main made use of the config object here. Can we bring that back? Rather than the loop below